test/functional/test-alp-handler.js

```diff
@@ -73,9 +73,8 @@ describe('alp-handler', () => {
     expect(open.callCount).to.equal(1);
     expect(open.lastCall.args).to.jsonEqual([
       'https://cdn.ampproject.org/c/www.example.com/amp.html#click=' +
-          'https%3A%2F%2Ftest.com%3Fadurl%3Dhttps%253A%252F%252F' +
-          'cdn.ampproject.org%252Fc%252Fwww.example.com%252Famp.html%26' +
-          'amp%3D1',
+          'https%3A%2F%2Ftest.com%3Famp%3D1%26adurl%3Dhttps%253A%252F%252F' +
```

Comment:
Since the producer of these strings was changed to add URL params to the front of the URL as opposed to the back, they are updated to reflect the new state.

In this particular case the URLs are encoded into other URLs which leads to double encoding, and which is the seeming middle of the URL.